src/OpenTelemetry/ProviderExtensions.cs

```diff
@@ -50,11 +50,11 @@ public static Resource GetResource(this BaseProvider baseProvider)
             return Resource.Empty;
         }
 
-        public static Func<Batch<Metric>> GetMetricCollect(this BaseProvider baseProvider)
+        public static Action GetCollectObservableInstruments(this BaseProvider baseProvider)
```

Comment:
This is changing the public API surface but this was only being called by `MetricReader.OnCollect`.

Reply:
The name confused me a first because it doesn't return the instruments, it returns a collection callback delegate. A few suggestions...

`GetObservableInstrumentCollect`
`GetObservableInstrumentCollectAction`
`GetObservableInstrumentCollectCallback`

Also looking at the diff it seems we only call this from `MetricReader` could we make it `internal`?

Reply:
`GetObservableInstrumentCollectCallback`